### PR TITLE
NAS-113903 / None / Input type="number" validation in Firefox

### PR DESCRIPTION
**Pull request by AlexKarpov98**
- Created a numberValidator()
- Shown validation errors of input type

### CODE REVIEW

File: src/app/modules/ix-forms/components/ix-input/ix-input.component.ts [hunk 1]
```diff
@@ -66,6 +71,10 @@ export class IxInputComponent implements ControlValueAccessor {
     this.onChange(this.value);
   }
 
+  inValidMessage(): string {
```
Comment (denysbutenko) [inline src/app/modules/ix-forms/components/ix-input/ix-input.component.ts:74]: The spelling looks weird. Could we have `invalid` and `invalidMessage`?

**Review by denysbutenko — CHANGES_REQUESTED** [review]
Kapture 2022-01-26 at 21 22 56
The clear button disappeared when input become invalid

File: src/app/modules/ix-forms/components/ix-input/ix-input.component.ts (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/modules/ix-forms/components/ix-input/ix-input.component.ts:74]: Sure!

File: src/app/modules/ix-forms/components/ix-input/ix-input.component.ts [hunk 2]
```diff
@@ -34,12 +35,14 @@ export class IxInputComponent implements ControlValueAccessor {
 
   isDisabled = false;
   showPassword = false;
+  inValid = false;
```
Comment (denysbutenko) [inline src/app/modules/ix-forms/components/ix-input/ix-input.component.ts:38]: `inValid` -> `invalid`

**Review by denysbutenko — APPROVED** [review]
image

Looks good

**Pull request merged**